staging/src/k8s.io/apiserver/pkg/endpoints/discovery/root.go

```diff
@@ -33,12 +34,18 @@ import (
 // GroupManager is an interface that allows dynamic mutation of the existing webservice to handle
 // API groups being added or removed.
 type GroupManager interface {
+	GroupLister
+
 	AddGroup(apiGroup metav1.APIGroup)
 	RemoveGroup(groupName string)
 
 	WebService() *restful.WebService
 }
 
+type GroupLister interface {
+	Groups(ctx context.Context, req *http.Request) ([]metav1.APIGroup, error)
```

Comment:
that's an odd signature. Can you comment what ctx and req are used for?

Reply:
I agree. I can get rid of req and move it back to the ServeHTTP flow. I was originally trying to make sure that we filled in the ServerAddressByClientCIDRs field everywhere we were returning discovery, including in our aggregation, but looking at what upstream does today, even that isn't consistent (some calls have it, some don't).

What do you think makes the most sense?

Reply:
am fine with either way, including just go-doc'ing what you just described about `ServerAddressByClientCIDRs`.

Reply:
New commit added - PTAL